src/pages/Home/Subscription.tsx

```diff
@@ -0,0 +1,80 @@
+import Header from '@/components/common/Header';
+import InputWithButton from '@/components/common/InputWithButton';
+import LoadingSpinner from '@/components/common/LoadingSpinner';
+import SubscriptionItem from '@/components/subscription/SubscriptionItem';
+import { SUBSCRIPTION_PAGE_STYLE } from '@/constants/styles';
+import { useState } from 'react';
+import AlertModal from '@/components/common/AlertModal';
+import { useSubscriptionsQuery } from '@/hooks/useSubscriptionsQuery';
+import useSubscribeMutation from '@/hooks/useSubscribeMutation';
+
+const Subscription = () => {
+  const [channelUrl, setChannelUrl] = useState('');
+  const [alertMessage, setAlertMessage] = useState('');
+
+  const { mutate: mutateSubscribe } = useSubscribeMutation(
+    () => setAlertMessage('구독 채널이 추가되었습니다.'),
+    (errorCode: string) => {
+      if (errorCode === '4402') {
+        setAlertMessage('이미 구독중인 채널입니다.');
+      } else if (errorCode === '4403') {
+        setAlertMessage('더 이상 채널을 구독할 수 없습니다.');
+      } else {
+        setAlertMessage('구독 채널 추가에 실패했습니다. URL을 확인해주세요.');
+      }
+    }
+  );
+
+  const { data, isLoading } = useSubscriptionsQuery();
```

Comment:
```suggestion
  const { data, isError, isLoading } = useSubscriptionsQuery();
  
  if (isError) {
      return <div> 데이터를 가져오지 못 했습니다 </div>
  }
```
만약 가져오는 쿼리가 문제가 생겼을 때, fallback 처리를 해야하지 않을까요??

Reply:
리뷰 감사합니다! 😊
fallback 처리 추가하면 좋을 것 같습니다.

@leesm0218 @smileboy0014 
추가로 useSuspenseQuery()를 사용하여 로딩과 에러를 로딩/에러 상태를 UI 컴포넌트 밖에서 공통적으로 처리할 수 있다고 합니다.
컴포넌트 내부 코드가 깔끔해지고, 관심사의 분리 측면의 장점도 생길 것 같네요.
```ts
<ErrorBoundary FallbackComponent={<Fallback />}> // 에러상태 분리
  <Suspense fallback={<Loading />}> // 로딩상태 분리
    <MyComponent />
  </Suspense>
</ErrorBoundary>
```
아래 관련 링크 달아두겠습니다!

### Ref
- https://ko.react.dev/reference/react/Suspense
- https://tanstack.com/query/latest/docs/framework/react/reference/useSuspenseQuery
- https://lasbe.tistory.com/204
- #23

Reply:
아 그렇게 많이 처리를 하는군요! 저걸 도입하면 공통으로 fallback 처리를 할 수 있어서 더 코드가 정말 깔끔해질 것 같습니다! 그럼 #23 이슈에 적용해서 테스크 진행하면 좋을 것 같습니다 😀